pyup/providers/github.py

```diff
@@ -12,13 +12,16 @@ class Provider(object):
     def __init__(self, bundle, integration=False):
         self.bundle = bundle
         self.integration = integration
+        self.__api = None
 
     @classmethod
     def is_same_user(cls, this, that):
         return this.login == that.login
 
     def _api(self, token):
-        return Github(token, timeout=50)
+        if not self.__api:
+            self.__api = Github(token, timeout=50)
```

Comment:
Need to check if **token** changed as well. In that case you would need to get new instance again.